File: 04-passwords/main.js
const generar = () => {
    let tam = prompt("¿Cuántos caracteres quieres que tenga tu contraseña?");
    let may = prompt("Escribe SI o NO, si quieres contemplar mayusculas");
    let num = prompt("Escribe SI o NO, si quieres contemplar números");
    let car = prompt("Escribe SI o NO, si quieres contemplar caracteres especiales");
    let password = "";

    let caracteres = 'abcdefghijklmnopqrstuvwxyz';
    if (may === "SI") {
        caracteres = caracteres + "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    }
    if (num === "SI") {
        caracteres = caracteres + "0987654321";
    }
    if (car === "SI") {
        caracteres = caracteres + '!"#$%&/()@';
    }

    for (let i = 0; i < tam; i++) {
        password = password + caracteres[Math.trunc(Math.random() * caracteres.length)]
    }

    return password;
}

generar();

File: 04-passwords/main.py
import random

def crearPassword(may,num,car,tam):    
	caracteres = list('abcdefghijklmnopqrstuvwxyz')
	if  may=='SI':
		caracteres.extend(list('ABCDEFGHIJKLMNOPQRSTUVWXYZ'))

	if num=='SI':
		caracteres.extend(list('0987654321'))

	if car=='SI':
		caracteres.extend(list('!"#$%&/()@'))

    password = ''

    for x in range(tam):
	    password += random.choice(caracteres)
        
    return password


print("Vamos a generar una contraseña para ti")
may = input("Escribe SI o NO quieres contemplar mayusculas")
num = input("Escribe SI o NO quieres contemplar números")
car = input("Escribe SI o NO quieres contemplar caracteres especiales")
tam = int(input("Escribe el tamaño que quieres tenga la contraseña"))

psw= crearPassword(may,num,car,tam)
print("la contraseña generada es = ", psw)